docs/concepts/architecture/cloud-controller.md

```diff
@@ -0,0 +1,249 @@
+
+In this document, the concepts behind the components of cloud controller manager, and its associated functions will be discussed in detail. Here's the architecture of a Kubernetes cluster without the Cloud Controller Manager
+
+![Pre CCM Kube Arch](https://i.imgur.com/LAmMku4.png)
+
+Design
+-------
+
+In the architecture diagram of a Kubernetes cluster above, Kubernetes and the Cloud integrate from many different components.Namely
+
+1. Kubelet
+2. Kube Controller Manager
+3. Kube ApiServer
+
+The CCM consolidates all of these cloud dependent logic from the above 3 components to create a single point of integration with the cloud. The new architecture with CCM looks like this:
+
+![CCM Kube Arch](https://i.imgur.com/iqiIFHY.png)
+
+Components of CCM
+------------------
+The CCM breaks away some of the functionality of Kube Controller Manager (KCM) and runs it as a separate process. Specifically, it breaks away those controllers in KCM that are cloud dependent. The KCM has the following cloud dependent controller loops:
+
+ 1. Node Controller
+ 2. Volume Controller
+ 3. Route Controller
+ 4. Service Controller
+
+The CCM currently (v1.8) runs the following controllers from the above list
+
+1. Node Controller
+2. Route Controller 
+3. Service Controller
+
+Additionally, it runs another controller called the PersistentVolumeLabels controller. This controller is responsible for setting the zone and region labels on PersistentVolumes created in GCE and AWS clouds. 
+
+To summarize,the CCM runs the following 4 controllers:
+
+1. Node Controller
+2. Route Controller 
+3. Service Controller
+4. Persistent Volume Labels Controller
+
+**Note:** Volume Controller was deliberately chosen to *NOT* be a part of CCM. Due to the complexity involved and due to the existing efforts to abstract away vendor specific volume logic, it was decided that Volume Controller will not be moved to CCM.   
+
+
+Functions of CCM
+------------------
+
+CCM inherits its functions from cloud provided dependednt components of Kubernetes. This section is structured based on the components from which CCM inherits its functions.
+
+### 1. Kube Controller Manager
+
+The majority of CCM's functions are derived from KCM. As mentioned in the previous section, CCM runs the following control loops
+
+1. Node Controller
+2. Route Controller 
+3. Service Controller
+4. Persistent Volume Labels Controller
+
+#### Node Controller
+
+The Node Controller is responsible for initializing a node by obtaining information about the nodes running in the cluster from the cloud provider. The node controller performs the following functions
+
+1. Initialize a node with cloud specific zone/region labels
+2. Initialize a node with cloud specific Instance details (type, size etc.)
+3. Obtain the node's network addresses (IP addresses, hostname etc.)
+4. In case a node becomes unresponsive, then it checks the cloud to see if the node has been deleted from the cloud. 
```

Comment:
Should mention it deletes the k8s node object when the instance no longer exists.